### PR TITLE
ERP-648 Target Hearing Date population

### PR DESCRIPTION
**Pull request by RababRabab**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ERP-648

### Change description ###

Populate Target Hearing date following changes to the Receipt date

**Does this PR introduce a breaking change?** (check one with "x")

[ ] Yes
[X] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java [hunk 1]
```diff
@@ -13,15 +14,18 @@
...
         List<String> errors = new ArrayList<>();
 
-        LocalDate dateOfReceipt = LocalDate.parse(receiptDate);
+        LocalDate dateOfReceipt = LocalDate.parse(caseData.getReceiptDate());
 
         if (dateOfReceipt.isAfter(LocalDate.now())) {
             errors.add(FUTURE_RECEIPT_DATE_ERROR_MESSAGE);
         }
+        else{
+            caseData.setTargetHearingDate(dateOfReceipt.plusDays(TARGET_HEARING_DATE_INCREMENT).toString());
```
Comment (jacarrete) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java]: I think if you don't return the caseData the setTargetHearingDate will not be populated

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java [hunk 2]
```diff
@@ -216,7 +216,7 @@ public CaseActionsForCaseWorkerController(CaseCreationForCaseWorkerService caseC
         DefaultValues defaultValues = getPostDefaultValues(ccdRequest.getCaseDetails());
         log.info("Post Default values loaded: " + defaultValues);
         CaseData caseData = defaultValuesReaderService.getCaseData(ccdRequest.getCaseDetails().getCaseData(), defaultValues, false);
-        errors = eventValidationService.validateReceiptDate(ccdRequest.getCaseDetails().getCaseData().getReceiptDate());
+        errors = eventValidationService.validateReceiptDate(caseData);
```
Comment (jacarrete) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java]: We could check the errors before actually doing any changes in the caseData.
errors = validateReceiptData
if errors then return the payload
if not then do the getPostDefaultValues and getCaseData

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java (hunk 1 above)
Comment (RababRabab) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java]: I have tested this and double checked now it, i can cee the target hearing date being updated and displayed on the case details tab (as per image i uploaded into Jira). The casedata is passed byref is data updated should be availabe for the Response.

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java (hunk 2 above)
Comment (RababRabab) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java]: noted. I will move the validation call above the defaultvalues code.

Comment (RababRabab) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java]: refactored in brach : ERP-216-Date-Returned-From_Judge-Validation

**Pull request merged**